driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1593,10 +1582,35 @@ else if (host.state == Host.State.DOWN)
             }
         }
 
+        private void connect() {
+            try {
+                controlConnection.connect();
+            } catch (UnsupportedProtocolVersionException e) {
+                ProtocolVersion newVersion = e.getServerVersion();
+                if (newVersion == e.getUnsupportedVersion()) {
+                    newVersion = newVersion.getLowerSupported();
+                    if (newVersion == null) {
+                        throw e;
```

Comment:
This was a bit hard to follow for me. I'm pushing a proposal with intermediate variables and comments.